### PR TITLE
ContextualMenu: fix submenu not showing updated props correctly

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

Fixed an issue where the submenu would not show updated items, even if they were passed through props.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 1]
```diff
@@ -231,8 +228,8 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
                 </ul>
               </FocusZone>
             ) : (null) }
-            { submenuProps ? ( // If a submenu properities exists, the submenu will be rendered.
-              <ContextualMenu { ...submenuProps } />
+            { this.state.expandedMenuItemKey ? ( // If a submenu properities exists, the submenu will be rendered.
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: could you update the comment? or remove it?

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 2]
```diff
@@ -479,33 +476,48 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
...
         expandedMenuItemKey: item.key,
-        submenuProps: submenuProps,
+        submenuTarget: target
       });
     }
   }
 
+  private _getSubmenuProps() {
+    let { submenuTarget, expandedMenuItemKey } = this.state;
+    let item = this._findItemByKey(expandedMenuItemKey);
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: `const`

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 3]
```diff
@@ -479,33 +476,48 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
...
+
+    if (item.subMenuProps) {
+      assign(submenuProps, item.subMenuProps);
+    }
+    return submenuProps;
+  }
+
+  private _findItemByKey(key: string): IContextualMenuItem {
+    let { items } = this.props;
+    for (let i = 0; i < this.props.items.length; i++) {
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: `for(const item of items) { .... }`

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: I don't think that this works, it only iterates off strings I think.

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: It looks like `item` in this case equals the index of item. So I could rewrite it as 
``` 
for(const item in items) {
  let it=items[item] ... ```

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: image

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: image

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: I think I'm going to leave this as is.

**Comment by cschleiden** [comment]
For of, not for in. That works.

On May 8, 2017 5:58 PM, Jon Schectman <notifications@github.com> wrote:

@joschect commented on this pull request.

________________________________

In packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx<https://github.com/OfficeDev/office-ui-fabric-react/pull/1745#discussion_r115388165>:

> +      id: this.state.subMenuId,
+      shouldFocusOnMount: true,
+      directionalHint: getRTL() ? DirectionalHint.leftTopEdge : DirectionalHint.rightTopEdge,
+      className: this.props.className,
+      gapSpace: 0
+    };
+
+    if (item.subMenuProps) {
+      assign(submenuProps, item.subMenuProps);
+    }
+    return submenuProps;
+  }
+
+  private _findItemByKey(key: string): IContextualMenuItem {
+    let { items } = this.props;
+    for (let i = 0; i < this.props.items.length; i++) {


I don't think that this works, it only iterates off strings I think.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx (hunk 3 above)
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: Why didn't I know about this! That's awesome. Thanks Chris!

**Pull request merged**